From 4679aa04c79abddfe3683d3cf4678b2f43118ed3 Mon Sep 17 00:00:00 2001
From: Brian Wood <brian.j.wood@intel.com>
Date: Tue, 24 Mar 2015 14:00:52 -0700
Subject: [PATCH] Update fdp nfcc driver to dev_pm_ops from legacy
 suspend/resume

This commit updates the fdp nfcc driver to the current
suspend/resume callbacks using the dev_pm_ops structure
for power management.

Signed-off-by: Brian Wood <brian.j.wood@intel.com>
---
 drivers/staging/fdp/fdp_custom.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fdp/fdp_custom.c b/drivers/staging/fdp/fdp_custom.c
index 4760061a5caf..497d2ab02f87 100644
--- a/drivers/staging/fdp/fdp_custom.c
+++ b/drivers/staging/fdp/fdp_custom.c
@@ -894,16 +894,20 @@ static const struct acpi_device_id fdp_acpi_match[] = {
 
 MODULE_DEVICE_TABLE(i2c, fdp_id);
 
+static const struct dev_pm_ops fdp_i2c_pm_ops = {
+	.suspend = fdp_suspend,
+	.resume = fdp_resume,
+};
+
 static struct i2c_driver fdp_i2c_driver = {
 	.driver = {
 		   .owner = THIS_MODULE,
 		   .name = I2C_ID_NAME,
 		   .acpi_match_table = ACPI_PTR(fdp_acpi_match),
+		   .pm = &fdp_i2c_pm_ops,
 	},
 	.probe = fdp_probe,
 	.remove = __devexit_p(fdp_remove),
-	.suspend = fdp_suspend,
-	.resume = fdp_resume,
 	.id_table = fdp_id,
 
 };
-- 
2.0.3

